app/models/note.rb

```diff
@@ -0,0 +1,13 @@
+# frozen_string_literal: true
+
+class Note < ApplicationRecord
+  validates :body, presence: true
+  belongs_to :submission
+  belongs_to :author, foreign_key: :created_by, class_name: "User", inverse_of: :authored_notes
```

Comment:
rubocop made me declare `inverse_of` on only the `author` field, not the submission field, which is weird because a) we don't have a use case for a user's `authored_notes` and b) it didn't require it on the submission association above, which we _do_ need.

Reply:
The `inverse_of` option could be quite dangerous because it's a way of making both sides of the relation circularly dependent of each other... I'd just disable the cop.

Reply:
FWIW, I'm not sure if `dependent: :nullify` is what we want here. I recall discussing (but can't find where we wrote down), that if a User A creates Notes B and C, and then User A gets deleted, we'll likely still want the orphaned foreign key on Notes B and C present, so that it's possible to at least know the shadow of a User (instead of loosing all attribution of a Note to a User).